mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb

```diff
@@ -391,3 +391,277 @@ func TestFrameworkProvider_LoadAndValidateFramework_credentials(t *testing.T) {
 // 		}
 // 	})
 // }
+
+func TestFrameworkProvider_LoadAndValidateFramework_region(t *testing.T) {
+
+	// Note: In the test function we need to set the below fields in test case's fwmodels.ProviderModel value
+	// this is to stop the code under test experiencing errors, and could be addressed in future refactoring.
+	// - Credentials: If we don't set this then the test looks for application default credentials and can fail depending on the machine running the test
+	// - ImpersonateServiceAccountDelegates: If we don't set this, we get a nil pointer exception ¯\_(ツ)_/¯
+
+	cases := map[string]struct {
+		ConfigValues              fwmodels.ProviderModel
+		EnvVariables              map[string]string
+		ExpectedDataModelValue    basetypes.StringValue
+		ExpectedConfigStructValue basetypes.StringValue
+		ExpectError               bool
+	}{
+		"region value set in the provider config is not overridden by ENVs": {
+			ConfigValues: fwmodels.ProviderModel{
+				Region: types.StringValue("region-from-config"),
+			},
+			EnvVariables: map[string]string{
+				"GOOGLE_REGION": "region-from-env",
+			},
+			ExpectedDataModelValue:    types.StringValue("region-from-config"),
+			ExpectedConfigStructValue: types.StringValue("region-from-config"),
+		},
+		// This test currently fails - PF code doesn't behave like SDK code
+		// TODO(SarahFrench) - address https://github.com/hashicorp/terraform-provider-google/issues/15714
+		// "region values can be supplied as a self link": {
+		// 	ConfigValues: fwmodels.ProviderModel{
+		// 		Region: types.StringValue("https://www.googleapis.com/compute/v1/projects/my-project/regions/us-central1"),
+		// 	},
+		// 	ExpectedDataModelValue:    types.StringValue("https://www.googleapis.com/compute/v1/projects/my-project/regions/us-central1"),
+		// 	ExpectedConfigStructValue: types.StringValue("us-central1"),
+		// },
+		"region value can be set by environment variable: GOOGLE_REGION is used": {
+			ConfigValues: fwmodels.ProviderModel{
+				Region: types.StringNull(),
+			},
+			EnvVariables: map[string]string{
+				"GOOGLE_REGION": "region-from-env",
+			},
+			ExpectedDataModelValue:    types.StringValue("region-from-env"),
+			ExpectedConfigStructValue: types.StringValue("region-from-env"),
+		},
+		"when no region values are provided via config or environment variables, the field remains unset without error": {
+			ConfigValues: fwmodels.ProviderModel{
+				Region: types.StringNull(),
+			},
+			ExpectedDataModelValue:    types.StringNull(),
+			ExpectedConfigStructValue: types.StringNull(),
+		},
+		// Handling empty strings in config
+		// TODO(SarahFrench) make these tests pass to address: https://github.com/hashicorp/terraform-provider-google/issues/14255
```

Comment:
Just confirming: the TODO's aren't to modify the tests themselves, but the framework logic that they are testing?

Reply:
Yes that's right - I'm planning to make the plugin framework code behave in a more SDK-like way so those tests pass, and then as part of 5.0.0 I'll deliberately switch the provider _back_ so that empty strings aren't ignored in configs. So undoing a breaking change accidentally added in 4.x.x and then intentionally re-adding it in 5.0.0.